manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.7.yaml

```diff
@@ -6,6 +6,8 @@ metadata:
   name: metadata-exchange-1.7{{- if not (eq .Values.revision "") }}-{{ .Values.revision }}{{- end }}
   {{- if .Values.meshConfig.rootNamespace }}
   namespace: {{ .Values.meshConfig.rootNamespace }}
+  {{- else if .Values.global.istioNamespace }}
+  namespace: {{ .Values.global.istioNamespace }}
```

Comment:
I don't think this fix the original issue, `{{ .Release.Namespace }}`, looks like the default value for `meshConfig.rootNamespace` is always `istio-system`.

Reply:
See: https://github.com/istio/istio/blob/master/manifests/charts/istio-control/istio-discovery/values.yaml#L166

Reply:
I couldn't find from where this value `{{ .Release.Namespace }}` is set. Maybe removing default value for `meshConfig.rootNamespace` will work. I will give it a try.

Reply:
> looks like the default value for meshConfig.rootNamespace is always istio-system

Yes, and it always assumes default value and else does not execute.


> In that case we have to set two extra namespaces vis old helm values:
> ```
> apiVersion: install.istio.io/v1alpha1
> kind: IstioOperator
> metadata:
>   namespace: istio-system-test
>   name: "example"
> spec:
>   namespace: istio-system-test
>   values:
>     global:
>       istioNamespace: istio-system-test
>     meshConfig:
>       rootNamespace: istio-system-test
> ```

@morvencao Does that means both the flags `global.istioNamespace` and `meshConfig.rootNamespace`  needs to be set?

Reply:
Yes.
We may need figure out how `namespace` field in iop spec maps to `istioNamespace` and `rootNamespace` if we donit expect  to set two extra namespaces vis old helm values.

Reply:
Thanks, I will look into it.